src/mo_frontend/effect.ml

```diff
@@ -25,13 +25,32 @@ let typ phrase = phrase.note.note_typ
 let eff phrase = phrase.note.note_eff
 
 let is_triv phrase  =
-    eff phrase = T.Triv
+  eff phrase = T.Triv
+
+let is_shared_func exp =
+  T.(match typ exp with
+    | Func (Shared _, _, _, _, _) -> true
+    | _ -> false)
+
+let is_local_async_func exp =
```

Comment:
Same helper functions seem to occur in `effect.ml`, `ir_effect.ml`, and `await.ml`. Could this be shared in a common place?

Reply:
Good catch!

Reply:
Done in #3697